### DISCUSSION TITLE
When redirecting with Apache, it goes through non-SSL

### DISCUSSION THREAD

**Discussion by riku22** [caprover/caprover#1933] [Q&A]
Hello.

I set up CapRover and deployed the site using Apache's images.
This site has forced SSL enabled.

This site has several redirects defined in `.htaccess`, but when redirecting, it goes through `http` once.
I think it would be better if you could redirect to `https` without going through `http`.
I think it's probably because Apache doesn't recognize it as an SSL site, but is there any workaround?

The contents of `.htaccess` are as follows.

`redirect 301 /page1.html /page2.html`

The execution result of `curl` is as follows.

```
$ curl -LI https://myapp.example.com/page_1.html

HTTP/2 301
server: nginx
date: Fri, 03 Nov 2023 00:03:58 GMT
content-type: text/html; charset=iso-8859-1
location: http://riku22.net/page_2.html

HTTP/1.1 302 Found
Server: nginx
Date: Fri, 03 Nov 2023 00:03:58 GMT
Content-Type: text/html
Content-Length: 162
Connection: keep-alive
Location: https://riku22.net/page_2.html

HTTP/2 200
server: nginx
date: Fri, 03 Nov 2023 00:03:58 GMT
content-type: text/html; charset=UTF-8
```

Sincerely.

**Comment by maietta** [comment]
This is not a CapRover specific issue, so your post belongs elsewhere.

However, I will say that you need to make sure you are using the correct syntax to get what you want.

Something like this should remove that second redirect.

```
RewriteEngine On
RewriteRule ^page1\.html$ https://yourdomain.com/page2.html [R=301,L]
```

**Reply by githubsaturn** [reply]
Nick's correct. Specifically, you may want to look into this:
https://stackoverflow.com/questions/19372594/preserve-http-https-protocol-in-htaccess-redirects

**Comment by riku22** [comment]
Sorry, I solved it myself.
This time, I was only thinking of redirects, so it looks like I just need to add `proxy_redirect http:// $scheme://;` to my nginx configuration.
In my case, this seems to be fine for now.
I apologize for any inconvenience caused to you.

Sincerely.
